tests/test.hs

```diff
@@ -28,6 +29,37 @@ main = defaultMain $ testGroup "Tests"
       files <- findByExtension [".c", ".h"] basedir
       sort files @?= (sort . map (basedir </>))
         ["d1/d2/h1.c","d1/g1.c","f1.c","f2.h"]
+  , testGroup "Missing golden files"
+    -- Make sure that each entrypoint to tasty-golden can properly create
+    -- golden files if they are not provided. This serves as a regression test
+    -- for #32.
+    [ goldenVsFile
+        "goldenVsFile without golden file"
+        "tests/golden/goldenVsFile.golden"
+        "tests/golden/goldenVsFile.actual"
+        (touch "tests/golden/goldenVsFile.actual")
+    , goldenVsFileDiff
+        "goldenVsFileDiff without golden file"
+        diff
+        "tests/golden/goldenVsFileDiff.golden"
+        "tests/golden/goldenVsFileDiff.actual"
+         (touch "tests/golden/goldenVsFileDiff.actual")
+    , goldenVsString
+        "goldenVsString without golden file"
+        "tests/golden/goldenVsString.golden"
+        (return "")
+    , goldenVsStringDiff
+        "goldenVsStringDiff without golden file"
+        diff
+        "tests/golden/goldenVsStringDiff.golden"
+        (return "")
+    , after AllFinish "$2 == \"* without golden file\"" $
```

Comment:
I think the correct syntax would be `$2 ~ /without golden file/`.

Reply:
An even better solution for this use case would be to use resources, I think, since `"Clean up tests/golden/goldenVs*.golden files"` is not a real test case but just a cleanup action (IIUC).

Reply:
Wow, I can't believe I didn't know about resources! Yes, that's a much better way to deal with this. I've pushed a new version that uses resources accordingly here.